src/main/java/net/consensys/wittgenstein/protocols/ETHPoW.java

```diff
@@ -231,11 +247,15 @@ private void mine1ms() {
     @Override
     public boolean onBlock(POWBlock b) {
       boolean res = super.onBlock(b);
+
       if (res) {
         // Someone sent us a new block, so we're going to switch
         //  our mining to this new head
         inMining = null;
       }
+      if (this.blocksReceivedByBlockId.put(b.id, b) != null) {
+        this.uncles.add(b);
+      }
```

Comment:
As well i don't see a check on
- can it be uncle or not (the right height is not enought)
- only 2 uncles
+ the impact of the uncles on the total difficulty (plus a unit test)

Reply:
when I create a new block with the constructor you pass the list of uncles but filter to only get up to 2 uncles.